UniversityTimetablingSystem/backend/university_timetabling/timetable/views.py

```diff
@@ -1,156 +1,330 @@
+timeslots_collection = db.collection(University.TIME_SLOTS)
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+project_root = os.path.abspath(os.path.join(current_dir, os.pardir, os.pardir, os.pardir, 'build'))
+sys.path.append(project_root)
+
+try:
+    import university_bindings as ub
+except ImportError as e:
+    raise ImportError(f"Failed to import university_bindings: {e}")
+
+university = ub.University()
+
 
 @csrf_exempt
 def add_instructor(request):
     if request.method != 'POST':
         return HttpResponseBadRequest('Invalid request method')
     try:
         data = json.loads(request.body)
-        name = data.get('name')
-        preferred_courses_data = data.get('preferredCourses', [])
-        availability_data = data.get('availability', [])
+        name = data.get(University.NAME)
+        preferred_courses_data = data.get(University.PREFERRED_COURSES, [])
+        availability_data = data.get(University.AVAILABILITY, [])
 
         if not name:
             return HttpResponseBadRequest('Missing name')
 
-        args = [
-            os.path.join(project_root, 'UniversityTimetablingSystem'),
-            '--addInstructor',
-            name,
-            '--preferredCourses',
-        ]
+        availability = process_availability(availability_data)
+        if availability is False:
+            return HttpResponseBadRequest('Invalid availability data')
 
-        for course in preferred_courses_data:
-            args.extend([course['courseName']])
+        preferred_courses = process_preferred_courses(preferred_courses_data)
+        if preferred_courses is False:
+            return HttpResponseBadRequest('Invalid preferred courses data')
 
-        args.extend(['--availability'])
-        for slot in availability_data:
-            args.extend([slot['day'], slot['startTime'], slot['endTime']])
+        if instructor_exists(name):
+            return HttpResponseBadRequest('Instructor with this name already exists')
 
-        result = subprocess.run(args, capture_output=True, text=True, check=True)
-        print('Output:', result.stdout)
+        create_and_add_instructor(name, preferred_courses, availability)
 
         return JsonResponse({'status': 'Instructor added successfully'})
-    except subprocess.CalledProcessError as e:
-        print(f"Error occurred: {e}")
-        return HttpResponseBadRequest(f"Error occurred: {e}")
     except Exception as e:
         print(f"Error occurred: {e}")
-        return HttpResponseBadRequest(f"Error occurred: {e}")
+        return HttpResponseServerError(str(e))
+
+
+def process_availability(availability_data):
+    availability = []
+    for slot in availability_data:
+        time_slot = {
+            University.DAY: slot[University.DAY],
+            University.START_TIME: slot[University.START_TIME],
+            University.END_TIME: slot[University.END_TIME]
+        }
+        if not all(time_slot.values()):
+            return False
```

Comment:
It's somehow strange that the function returns either list or Boolean. It's better to return None for such cases. Change other places too if there are any